.circleci/config.yml

```diff
@@ -29,6 +29,9 @@ templates:
         - run:
             name: Build & run tests
             command: .circleci/test.sh
+        - run:
+            name: Upload code coverage report
+            command: .circleci/codecov.sh
```

Comment:
Do we? This runs within the same sociomantic docker container as beaver uses. I can of course pack `lst` files as artifacts and submit them in a separate stage but not sure if it is necessary.

Reply:
I mean, eventually we'll probably need to add credentials, which could be available for this untrusted script. But I'm not sure how circleci deals with credentials. If there is any way to make sure the script doesn't get them, that would be enough.

Reply:
You can explicitly override some of env variables for specific steps - https://circleci.com/docs/2.0/env-vars/#adding-environment-variables-to-a-step

Reply:
OK, let's see how it goes. If we are going to consider running an untrusted script safe enough, then I'm not sure it makes sense to save a copy of the script.

Reply:
I know it was my idea in the first place, and that we can potentially audit the script and trust it, I don't think it's something that will ever happen in practice, so why bother... I would try to remove as much access as possible to the script though. For example, how are credentials Git/GitHub handled by circleci? If you do `git push` inside circle, can you push to the repo? That would be very bad when running an untrusted script.

Reply:
I did it for performance reasons mostly - why download something on each build if it can be trivially stored instead?

Reply:
It is downloaded anyway when you check out the code :)
But I care more about scale, if we copy this script to every repo out there it means every repo is responsible for it, meaning they need to care individually about when it is needed to upgrade and it pollutes the repo. I don't think the repo is the place to store the script. (since this is an experiment I'm fine as something temporary while we test).

Anyway, my main concern is about security. I want to be convinced the script is sufficiently confined (including env vars and git credentials)

Reply:
https://discuss.circleci.com/t/marking-specific-job-step-as-untrusted/18565